topdown/cache/cache.go

```diff
@@ -96,11 +148,26 @@ type cache struct {
 	mtx    sync.Mutex
 }
 
+func newCache(config *Config) *cache {
+	return &cache{
+		items:  map[string]cacheItem{},
+		usage:  0,
+		config: config,
+		l:      list.New(),
+	}
+}
+
 // Insert inserts a key k into the cache with value v.
-func (c *cache) Insert(k ast.Value, v InterQueryCacheValue) (dropped int) {
+func (c *cache) InsertWithExpiry(k ast.Value, v InterQueryCacheValue, expiresAt time.Time) (dropped int) {
```

Comment:
Nit: The description of the function needs to be updated to reflect the new name for example.